### DISCUSSION TITLE
git-recieve-pack not permitted using eclipse

### DISCUSSION THREAD

**Discussion by NateLott** [community/community#64553] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

I'm trying to "push branch to master" on Eclipse but when I put in my github login info I get "Can't connect to any URL: .....  and fit-recieve-pack not permitted. How does one go on to fix this issue?

**Comment by Kaysuto** [comment]
Hello!

It seems the issue is due to git access permissions on Eclipse. The "git-receive-pack not permitted" error indicates that Eclipse is not allowed to push changes to the remote GitHub repository.

Some things to check to resolve this:

- Verify you are using the correct GitHub username and password credentials in Eclipse.
- In Eclipse's Git preferences, add a new entry in "Configured Credentials Storage" with your GitHub personal access token.
- Make sure SSH access is set up between Eclipse and GitHub and the SSH key is added to your GitHub account.
- Check the permissions on your GitHub repository - your account needs write access to push changes.
- Remove any existing GitHub credential files in the local .git/ directory of your project.
- As an alternative, try pushing changes via the git command line instead of Eclipse.
- As a last resort, delete and re-clone your local repository to reset credentials.

Let me know if you have any other questions! I'm happy to help further troubleshoot the issue.

**Reply by NateLott** [reply]
What if I'm using HTTPS?

**Comment by Kaysuto** [comment]
Hello,

Based on the "**git-receive-pack not permitted**" error, it seems the issue is due to lacking push permissions to the GitHub repository.

In newer GitHub updates, applications are not allowed direct access to repository accounts. To overcome this, you need to **create** a personal access token from "**Settings** > **Developer settings** > **Personal access tokens**" with repo permissions.

Then use this token instead of your password in Eclipse or other applications. **This should grant the needed permissions to push.**

Some things to try:

- Generate a new token and configure it in Eclipse's Git settings.
- Verify the token has the necessary repo access scopes enabled.
- Test pushing from Eclipse after entering the token.
- If still not working, try pushing from the git CLI using the token to isolate the issue.
- Check that SSH keys are set up properly between Eclipse and GitHub.
- As a last resort, delete and recreate the remote to reset all configs.

The solutions about **password/auth** may not help since this is due to the **recent** GitHub permission changes. Using a **personal access token** should grant Eclipse the access it needs. Let me know if the problem persists! 😉

**Comment by arafajs** [comment]
Try VPN.

**Comment by Kaysuto** [comment]
Or watch this: https://youtu.be/rLEYw1caoBQ

**Comment by DINESHKUMAR-CA** [comment]
None of the solutions worked for me. I am still facing the same issue.

**Reply by AmatyaSumit** [reply]
same issue here too. Are you trying to push it in your own repo or some one else's?

**Reply by DINESHKUMAR-CA** [reply]
I tried to push the code to my repository and have followed everything mentioned by **Kaysuto**. But now I have resolved the issue and successfully uploaded my code to my repository. I found an issue with my tokens.

**Reply by defectio** [reply]
I have a same issue too, could you let me know what is the issue of your token?
